extensions/aproc/proc/dc3build/dc3build_process.py

```diff
@@ -0,0 +1,274 @@
+                                             EVENT_OUTCOME_KEY, EVENT_REASON,
+                                             EVENT_TYPE_KEY, USER_ACTION_KEY,
+                                             USER_EMAIL_KEY, USER_ID_KEY)
+
+DRIVERS_CONFIGURATION_FILE_PARAM_NAME = "drivers"
+LOGGER = Logger.logger
+
+
+class OutputDC3BuildProcess(BaseModel):
+    item_location: str = Field(title="Item location", description="Location of the Item on the ARLAS Item Registration Service")
+    id: str = Field(title="Identifier", description="Identifier of the cube item within the collection")
+    collection: str = Field(title="Collection", description="Collection that contains the item")
+    catalog: str = Field(title="Catalog", description="Catalog that contains the item")
+
+
+summary: ProcessSummary = ProcessSummary(
+    title="Build a cube based on catalog items, then register the result in ARLAS.",
+    description="Build a data cube (time serie of ) based on groups of items, each group representing a time slice. The result is register in ARLAS",
+    keywords=["cube", "build", "datacube", "dc3build"],
+    id="dc3build",
+    version="0.1",
+    jobControlOptions=[JobControlOptions.async_execute],
+    outputTransmission=[TransmissionMode.reference],
+    # TODO: provide the links if any => link could be the execute endpoint
+    links=[]
+)
+
+description: ProcessDescription = ProcessDescription(
+    **summary.model_dump(exclude_none=True, exclude_unset=True),
+    inputs=base_model2description(InputDC3BuildProcess),
+    outputs=base_model2description(OutputDC3BuildProcess)
+)
+
+
+class AprocProcess(Process):
+
+    @staticmethod
+    def init(configuration: dict):
+        if configuration.get(DRIVERS_CONFIGURATION_FILE_PARAM_NAME):
+            InputDC3BuildConfiguration.init(configuration_file=configuration.get(DRIVERS_CONFIGURATION_FILE_PARAM_NAME))
+            InputDC3BuildConfiguration.raise_if_not_valid()
+            DriverManager.init(summary.id, InputDC3BuildConfiguration.settings.drivers)
+        else:
+            raise DriverException("Invalid configuration for enrich drivers ({})".format(configuration))
+        AprocProcess.input_model = InputDC3BuildProcess
+        description.inputs.get("include_drivers").schema_.items.enum = DriverManager.driver_names(summary.id)
+        description.inputs.get("exclude_drivers").schema_.items.enum = DriverManager.driver_names(summary.id)
+
+        AccessManager.init()
+
+    @staticmethod
+    def get_process_description() -> ProcessDescription:
+        return description
+
+    @staticmethod
+    def get_process_summary() -> ProcessSummary:
+        return summary
+
+    @staticmethod
+    def __get_driver__(input: InputDC3BuildProcess) -> DC3Driver:
+        driver: DC3Driver = None
+        if input.composition and len(input.composition) > 0 and len(input.composition[0].dc3__references):
+            reference = input.composition[0].dc3__references[0]
```

Comment:
is there a chance that reference is None in case `input.composition[0].dc3__references` is empty or undefined ?

Reply:
could be. I add a test.